### PR TITLE
Add support for RINGOVD section in pnor

### PR DESCRIPTION
**Pull request by pragupta**
This change is [Reviewable](https://reviewable.io/reviews/open-power/pnor/59)

### CODE REVIEW

**Comment by williamspatrick** [comment]
Why is this only in the Layout_64 image?

Why do we even have a ring override section?  Doesn't that violate secureboot?  @bofferdn

**Comment by williamspatrick** [comment]
And why is this not in the P9-specific part of the script?  Won't it break P8 systems?

**Comment by pragupta** [comment]
Added the section to all the p9Layouts and moved it to p9 specific part of the script. @bofferdn said that RINGOVD won't be used if we are in booting in secure mode.

**Comment by stewart-ibm** [comment]
Prachi Gupta <notifications@github.com> writes:
> Added the section to all the p9Layouts and moved it to p9 specific
> part of the script. @bofferdn said that RINGOVD won't be used if we
> are in booting in secure mode.

How many partitions does hostboot now scan for overrides?

ATTR_TMP, ATTR_PERM, HBEL?, RINGOVD, GARD.

That's 4 or 5.

That's getting insane, and a complete waste of PNOR space as each of
these are statically allocated.

I'd argue that what should happen is that there's *one* format that
allows temporary or permanent overrides of sensible defaults in a
standard format. Say, like the NVRAM partition does for
skiboot/petitboot/linux.

-- 
Stewart Smith
OPAL Architect, IBM.

**Comment by dcrowell77** [comment]
The data in RINGOVD is not even remotely similar to the data in ATTR_TMP/ATTR_PERM.  This partition holds raw scanring data that is consumed by the SBE/HCODE logic to change the hardware initialization.  Trying to squeeze that into the attribute structure would be time-consuming and confusing to all of the users.  Having separate sections makes this infinitely more user-friendly for the HW team (the primary users).  The format of this data is completely opaque to us so that it matches with the data that is already included inside the HW ref images.

RINGOVD is not *required* for boot.  If a system is short on PNOR space, it can be removed.  All that does is limit debug/test options in the lab.

Note, HBEL and GARD are not overrides, they are error logs and persistent deconfig records respectively.

**Comment by wghoffa** [comment]
@williamspatrick @stewart-ibm 

Are there still pending questions/comments to be addressed? Can you summarize those if so?
